core/src/main/java/org/apache/iceberg/util/ParallelIterable.java

```diff
@@ -257,17 +263,21 @@ private static class Task<T> implements Supplier<Optional<Task<T>>>, Closeable {
     @Override
     public Optional<Task<T>> get() {
       try {
+        if (queue.size() >= approximateMaxQueueSize) {
+          // Tasks might hold references (via iterator) to constrained resources
```

Comment:
Checking the code, this comment seems to be in conversation with the previous comment that was removed below, it also removes the context of the "queue / ApproximateMaxQueueSize" check.

I think it should definitely mention first that we "yield when the queue is over the size limit". The rest of this information isn't really relevant to this block of code without knowing that it previously used to allow for yielding in between elements of the iterator.